linux/Dockerfile

```diff
@@ -0,0 +1,12 @@
+FROM golang:1.14-stretch
+
+RUN apt-get -y update \
+    && apt-get -y install \
+        gettext-base \
+    && apt-get -y clean \
+    && rm -rf /var/lib/apt/lists/*
+
+ADD "generate_tar.sh" "."
+ADD "gcp-config.yaml" "."
+
+CMD ["./generate_tar.sh"]
```

Comment:
Usually the sepration I've seen is:

- Dockerfile includes all dependencies needed for building
- Build scripts are mounted and run inside the docker container

The dependencies change infrequently so you rarely need to rebuild your Dockerfile, whereas your build scripts are likely to change slightly more often.

That doesn't necessarily make as big of a difference as you would think though as each command in a Docker file gets cached, so if you only change the build scripts, only the most recent commands (3 in this Dockerfile) would need to be recreated.

But ideally you don't even want to have to rebuild your Dockerfile on each build.

Reply:
Since Go can be cross-compiled for any OS, I'm keen to keep this a bit simpler and use the existing Dockerfile in this repo for builds on all environments unless we find there's significantly different dependencies

Reply:
So based on my understanding, we want this dockerfile to only contain the setups of the environment. In this case, how should I go about it if I want this container to
1. pull down a repo
2. build the binary
3. compress files and folders into a tar file

Reply:
currently reusing the dockerfile in .build. mount and execute the script when running the container.